cmd/xl-v1-utils.go

```diff
@@ -284,32 +282,35 @@ func readXLMetaStat(disk StorageAPI, bucket string, object string) (si statInfo,
 	// obtain xlMetaV1{}.Stat using `github.com/tidwall/gjson`.
 	xlStat, err := parseXLStat(xlMetaBuf)
 	if err != nil {
-		return si, nil, errors2.Trace(err)
+		logger.LogIf(ctx, err)
+		return si, nil, err
 	}
 
 	// Return structured `xl.json`.
 	return xlStat, xlMetaMap, nil
 }
 
 // readXLMeta reads `xl.json` and returns back XL metadata structure.
-func readXLMeta(disk StorageAPI, bucket string, object string) (xlMeta xlMetaV1, err error) {
+func readXLMeta(ctx context.Context, disk StorageAPI, bucket string, object string) (xlMeta xlMetaV1, err error) {
 	// Reads entire `xl.json`.
 	xlMetaBuf, err := disk.ReadAll(bucket, path.Join(object, xlMetaJSONFile))
 	if err != nil {
-		return xlMetaV1{}, errors2.Trace(err)
+		logger.LogIf(ctx, err)
+		return xlMetaV1{}, err
 	}
 	// obtain xlMetaV1{} using `github.com/tidwall/gjson`.
-	xlMeta, err = xlMetaV1UnmarshalJSON(xlMetaBuf)
+	xlMeta, err = xlMetaV1UnmarshalJSON(ctx, xlMetaBuf)
 	if err != nil {
-		return xlMetaV1{}, errors2.Trace(err)
+		logger.LogIf(ctx, err)
```

Comment:
Possible duplicate logging.
`xlMetaV1UnmarshalJSON` is supposed to take care or already takes care of the `err` error condition.

Reply:
done